README.md

```diff
@@ -74,8 +75,9 @@ android {
         javaCompileOptions {
             annotationProcessorOptions {
                 arguments = [
-                    stagGeneratedPackageName: 'com.vimeo.sample.stag.generated',
-                    stagDebug               : 'true'
+                    stagAssumeHungarianNotation: 'true'
+                    stagGeneratedPackageName   : 'com.vimeo.sample.stag.generated',
+                    stagDebug                  : 'true'
```

Comment:
Because Gson isn't generating any code but just accessing fields at compiler time, we can't rely on something like a constructor or method parameter to determine whether to accept hungarian notation or not. We need to know at compile time what the name of the methods are so we can create code.

That leaves only two concrete solutions in my mind: using an annotation, or using a processor argument.

I do like the idea of an annotation level parameter, but my concern with that is that it could become a hassle to add the naming scheme used to every class declaration, especially when any reasonable codebase uses consistent naming.

A third solution that I've thought of is to remove the option to choose, and have stag choose for you. So, given a field `mField`, it would look for two different naming schemes, one with the first letter intact, and one without (for setters `setMField` and `setField`). This would mean no intervention from the consumer. I didn't implement it this way though because it's non deterministic and there's no way to know which method to choose if both `setMField` and `setField` exist in a class. In a small subset of scenarios, this might even generate the wrong code. Given two fields `mode` and `ode`, with an ambiguous naming scheme, `setOde` could justifiably be used to set either field. We could try to enforce consistent naming per class, but determining this seems problematic. We could remove some ambiguity by checking whether the first letter of a field starts with `m`, but idk.